config/config.go

```diff
@@ -18,8 +18,9 @@ type Config struct {
 
 // TargetConfig represents the configuration of a single Prometheus API endpoint.
 type TargetConfig struct {
-	QueryURL string `yaml:"query_url"`
-	TSDBPath string `yaml:"tsdb_path"`
+	QueryURL  string `yaml:"query_url"`
+	XQueryKey string `yaml:"x_query_key"`
```

Comment:
As mentioned above, I'd just call this `Headers` and allow passing arbitrary ones as a map.

Reply:
👍